src/backend/model_deployments/single_container.py

```diff
@@ -0,0 +1,76 @@
+import os
+from typing import Any, Dict, Generator, List
+
+import cohere
+from cohere.types import StreamedChatResponse
+
+from backend.chat.collate import to_dict
+from backend.model_deployments.base import BaseDeployment
+from backend.model_deployments.utils import get_model_config_var
+from backend.schemas.cohere_chat import CohereChatRequest
+from backend.services.metrics import (
+    collect_metrics_chat,
+    collect_metrics_chat_stream,
+    collect_metrics_rerank,
+)
+
+DEFAULT_RERANK_MODEL = "rerank-english-v2.0"
+SC_URL_ENV_VAR = "SINGLE_CONTAINER_URL"
+SC_MODEL_ENV_VAR = "SINGLE_CONTAINER_MODEL"
+SC_ENV_VARS = [SC_URL_ENV_VAR, SC_MODEL_ENV_VAR]
+
+
+class SingleContainerDeployment(BaseDeployment):
+    """Single Container Deployment."""
+
+    client_name = "cohere-toolkit"
+
+    def __init__(self, **kwargs: Any):
+        self.url = get_model_config_var(SC_URL_ENV_VAR, **kwargs)
+        self.model = get_model_config_var(SC_MODEL_ENV_VAR, **kwargs)
+        self.client = cohere.Client(
+            base_url=self.url, client_name=self.client_name, api_key="none"
```

Comment:
Should this be None?

Reply:
@BeatrixCohere iirc, the client complains if there's no `api_key` set?